test/functional/feature_taproot.py

```diff
@@ -1616,41 +1606,40 @@ def gen_test_vectors(self):
         self.block_submit(self.nodes[1], txn, "Crediting txn", None, sigops_weight=10, accept=True)
 
         # scriptPubKey computation
-        if GEN_TEST_VECTORS:
-            print("* Test vectors for scriptPubKey construction:")
-            for i, tap in enumerate(taps):
-                print("** Test case %i:" % (i + 1))
-                print("*** Given:")
-                print("**** Internal public key: <code>0x%s</code>" % tap.internal_pubkey.hex())
-                if script_descs[i] is None:
-                    print("**** Script tree: none")
+        tests = {}
+        spk_tests = tests.setdefault("scriptPubKey", [])
+        for i, tap in enumerate(taps):
+            test_case = {}
+            given = test_case.setdefault("given", {})
+            given['internalPubkey'] = tap.internal_pubkey.hex()
+            def pr(node):
+                if node is None:
+                    return None
+                elif isinstance(node, tuple):
+                    return {"id": int(node[0]), "script": node[1].hex(), "leafVersion": node[2]}
+                elif len(node) == 1:
+                    return pr(node[0])
+                elif len(node) == 2:
+                    return [pr(node[0]), pr(node[1])]
```

Comment:
Mentioned in the other PR.

```suggestion
            def pr(node):
                if node is None:
                    return []
                elif isinstance(node, tuple):
                    return {"id": int(node[0]), "script": node[1].hex(), "leafVersion": node[2]}
                elif len(node) == 1:
                    return [pr(node[0])]
                elif len(node) == 2:
                    return [pr(node[0]), pr(node[1])]
```